### ISSUE TITLE
pawn stuck in an endless get ammo loop.

### ISSUE THREAD

**Issue by SheakZarnn** [skyarkhangel/CombatRealism#72]
I have a stockpile with several stacks of 6x24mm charge cartridges of different types. One of my pawns, despite having his loadout set for normal ones, would infinitely run around trying to pick other ones and dropping them instantly. What's even more strange, I have 2 other pawns with the exact same loadout and they don't have this problem. It only appeared in last update (1.6.9.4)

**Comment by Thorbane** [comment]
How much mass can he carry?

**Comment by SheakZarnn** [comment]
Current weight 22kg, capacity 35 kg. Just to clarify some things: he had both charge rifle and ammo in his inventory (to match the loadout) and then infinitely tried to pick some more and dropped it 'cause it didn't match loadout. My guess, it is somehow connected to that new AI behaviour that makes them get ammo for their weapon.

**Comment by Gariba44** [comment]
This is probably an issue with the auto-ammo feature. Pawns are prioritizing "find any ammo for weapon" over "get loadout items". Once they pick up some ammo, if it's not on their loadout they immediately drop it and then go back to auto-ammo fill. This also happens if your loadout is set to carry less ammo then the auto-ammo threshold.
One solution for now is draft the colonist doing it, get him on top of the ammo you want him to pick up and undraft, since he will always look for the closest ammo.

**Comment by MGuffin** [comment]
I got exactly the same issue yesterday.

-A pawn had a survival rifle and 103 ammo (for this rifle) (no loadOut assigned)
-then I created a loadOut: Survival rifle + 30 ammo
-then, when I assigned the loadOut to that pawn loadout (Survival rifle + 30 ammo), the pawn dropped and retook all 103 ammos in an infinite loop.

I had unassign the loadout, then  force him to drop everything manually, then reassign the loadOut, for this to work.

**Comment by MGuffin** [comment]
I got the behavior again, and I think I have a better understanding of what happens:

-The pawn have a loadout assigned, a gun+ x Ammo
-the game orders that pawn to haul the same type of ammo to a storage space
-The pawn takes those ammo, then get more ammo in its backpack than it should so it drop them immediately
-Then the game asks it to haul the ammo again
etc. (loop)

**Comment by SheakZarnn** [comment]
No, to haul an item, pawn holds it in hands, not inventory. As I mentioned and Gariba confirmed, it is related to automatic ammo resupplying. Pawn tries to get ammo to fill bulk/mass capacity to a certain point automatically, and then checks the loadout. On the other hand, right now I'm playing without loadouts at all, and pawns almost never have issues with ammo.

**Comment by MGuffin** [comment]
It *should* hold it in is hands, but I looked closely, and the extra ammo is actually going in the inventory.
Not everytime, however.
I think this behavior starts when the ammo has the same caliber but not the same type (the pawn tries to haul a .303 hp, and needs .303ap in its loadout for example).
But yes the issue occurs when loadout are involved

**Comment by MGuffin** [comment]
Still getting this issue in  v.1.6.9.5 
My last tests are mostly confirming what Gariba44 and SheakZarnn said:
it look like a conflict between loadOut and autoRessuplying.

**Comment by skyarkhangel** [comment]
check with latest update in master tree

**Comment by MGuffin** [comment]
seems to be fixed, yes, thx!
